tests/integration/article.test.js

```diff
@@ -121,4 +123,61 @@ describe("# Articles endpoints", () => {
         });
     });
   });
+
+  describe("POST /articles/:slug/bookmark", () => {
+    it("should bookmark the article and return the success message", done => {
+      const { article } = createdArticle;
+      chai
+        .request(app)
+        .post(`/api/v1/articles/${article.slug}/bookmark`)
+        .set("Authorization", `Bearer ${token}`)
+        .end((err, res) => {
+          expect(res.status).equals(CREATED);
+          expect(res.body.message).to.contain("Article bookmarked");
+          expect(res.body).to.haveOwnProperty("bookmark");
+          expect(res.body.bookmark).to.be.an("object");
+          expect(res.body.bookmark).to.haveOwnProperty("createdAt");
+          expect(res.body.bookmark.articleId).equals(article.id);
+          done();
+        });
+    });
+
+    it("should deny bookmarking if no access-token provided", done => {
+      const { article } = createdArticle;
+      chai
+        .request(app)
+        .post(`/api/v1/articles/${article.slug}/bookmark`)
+        .end((err, res) => {
+          expect(res.status).equals(UNAUTHORIZED);
+          expect(res.body.message).to.contain("Please provide a token");
+          done();
+        });
+    });
+
+    it("should decline bookmarking the article which doesn't exist", done => {
+      const { article } = createdArticle;
+      chai
+        .request(app)
+        .post(`/api/v1/articles/${article.slug}fs1/bookmark`)
+        .set("Authorization", `Bearer ${token}`)
+        .end((err, res) => {
+          expect(res.status).equals(BAD_REQUEST);
```

Comment:
Is BAD_REQUEST the Correct error message here?

Reply:
Thanks, I have applied the feedback.